sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -784,6 +786,8 @@ public boolean advance() throws IOException {
               && rangeTracker.tryReturnRecordAt(true, ByteKey.of(reader.getCurrentRow().getKey()));
       if (hasRecord) {
         ++recordsReturned;
+      } else {
+        rangeTracker.markDone();
```

Comment:
up to you: could use this syntax from OffsetBasedSource instead.

https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/io/OffsetBasedSource.java#L271